pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1262,6 +1262,13 @@ func (vs *vSphereVMProvider) reconcileSnapshotRevert(
 		vmCtx.Logger.V(5).Info("No current snapshot specified, skipping revert")
 		return false, nil
 	}
+
+	// Skip snapshot revert processing for VKS/TKG nodes
+	if kubeutil.HasCAPILabels(vmCtx.VM.Labels) {
+		vmCtx.Logger.V(4).Info("Skipping snapshot revert for VKS/TKG node",
```

Comment:
This logger should already have the VM name. Is there like a condition we could report this on instead?

Reply:
Hmm.  If we report a condition like that, when would we clear it?  It will be a one time condition that will stick, which isn't nice?
And since the VKS VMs are always created with these labels, we would never get to this code anyway since the webhook would prevent the snapshot.  I added this as an additional safety check.

Reply:
Merging this after our conversation.  We can revisit this later if it turns out to be an issue.